.travis.yml

```diff
@@ -1,7 +1,20 @@
 language: java
-script: "mvn verify"
-jdk:
-  - openjdk8
+os: linux
+
+jdk: openjdk8
+
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+  - rm -fr $HOME/.gradle/caches/*/scripts/
+
 cache:
   directories:
     - "$HOME/.m2"
+    - "$HOME/.gradle/caches/"
+    - "$HOME/.gradle/wrapper/"
+
+script:
+  - rm -rf ~/.gradle/wrapper/dists/gradle-2.3-bin
+  - mvn clean install -DskipTests
+  - mvn verify
```

Comment:
What's the reason for `install` then `verify`?

Reply:
I was trying to resolve the build issues and found this commit (81adfe76acdca7b5e965350d02b3bd91a1675461) and it mentioned that CI tests might be broken unless do a build then do a test build.

@ben-manes Is this unnecessary now?

Reply:
This might be necessary with the eclipse-groovy-compiler. I will try removing it.

Reply:
That commit comment wrt CI failure was a one-time breakage and no longer relevant. Just required a clean if I recall correctly.

Reply:
@joelittlejohn I have already removed the extra build and haven't notice any problems.
@ben-manes Thanks for your input.

Reply:
So I think all the changes in this file can be reverted, is that right?

Reply:
I have removed the hacky gradle bin and double building stuff from Travis CI configuration but left in the caching.